rotkehlchen/tests/integration/test_eth2.py

```diff
@@ -386,9 +395,16 @@ def test_withdrawals_detect_exit(eth2: 'Eth2', database):
 @pytest.mark.freeze_time('2023-11-20 07:07:55 GMT')
 def test_query_no_withdrawals(
         eth2: 'Eth2',
+        database: 'DBHandler',
         ethereum_accounts: list[ChecksumEvmAddress],
 ) -> None:
     """Test that if an address has no withdrawals we correctly handle it"""
+    with database.user_write() as write_cursor:
+        # Add a validator associated with the address so that withdrawals get queried.
```

Comment:
Hmm this makes things tricky I just realized. How hard would it be to modify the logic to also query withdrawals and remember the exited or not if validator is not trcked but just withdrawal address is. Do not perform this change yet in this PR.

I am just thinking what it would look like. As for example it's possible to set as the withdrawal address of a validator an address of someone else and that someone else just getting withdrawals. For whatever reason. I am not sure how common that would be but it's in the realm of possibility.